drivers/spi/spi_nrfx_spim.c

```diff
@@ -600,21 +600,21 @@ static int spim_nrfx_pm_action(const struct device *dev,
 		())
 
 #ifdef CONFIG_SPI_0_NRF_SPIM
-SPI_NRFX_SPIM_DEVICE(0);
+SPI_NRFX_SPIM_DEFINE(0);
 #endif
 
 #ifdef CONFIG_SPI_1_NRF_SPIM
-SPI_NRFX_SPIM_DEVICE(1);
+SPI_NRFX_SPIM_DEFINE(1);
 #endif
 
 #ifdef CONFIG_SPI_2_NRF_SPIM
-SPI_NRFX_SPIM_DEVICE(2);
+SPI_NRFX_SPIM_DEFINE(2);
 #endif
 
 #ifdef CONFIG_SPI_3_NRF_SPIM
-SPI_NRFX_SPIM_DEVICE(3);
+SPI_NRFX_SPIM_DEFINE(3);
 #endif
 
 #ifdef CONFIG_SPI_4_NRF_SPIM
-SPI_NRFX_SPIM_DEVICE(4);
+SPI_NRFX_SPIM_DEFINE(4);
 #endif
```

Comment:
Could this explicit list be replaced with `DT_FOREACH_STATUS_OKAY` or something similar?

Reply:
> Could this explicit list be replaced with DT_FOREACH_STATUS_OKAY or something similar?

No; this is not possible due to limitations in nrfx.

Reply:
> not possible due to limitations in nrfx

@nika-nordic, would it be possible to alleviate this limitation?

Reply:
@hubertmis could be, but first I would need to understand those limitations. I presume this is because of how the nrfx instances are created, but could you share some more details @mbolivar-nordic ?

Reply:
> could you share some more details @mbolivar-nordic ?

This has been the subject of some internal meetings lately. I will forward you both a link to the minutes and you can discuss further with the responsible persons.

One basic issue, however, is that nrfx APIs like nrfx_spim_0_irq_handler() require knowing exactly which SPIM instance you are dealing with -- in this example, SPIM0. Devicetree instance numbers, by design, are interchangeable, and instance number 0 could be SPIM0 or SPIM3, with no guarantees made.

Therefore our use of nrfx here forces us to individually instantiate nodes based on node labels (`spi0` is SPIM0, etc.) so that we can correctly define IRQ handlers.